python/ray/scripts/scripts.py

```diff
@@ -741,82 +738,107 @@ def start(
+                    )
+                    cli_logger.newline()
+                    cli_logger.print(
+                        "`{}=true` must also be passed to any Ray clients.",
+                        ray_constants.ENABLE_RAY_CLUSTERS_ENV_VAR,
+                    )
+                cli_logger.newline()
+            else:
+                cli_logger.print("To add another node to this Ray cluster, run")
+                # NOTE(kfstorm): Java driver rely on this line to get the address
+                # of the cluster. Please be careful when updating this line.
                 cli_logger.print(
-                    "This Ray runtime only accepts connections from local host."
+                    cf.bold("  ray start --address='{}'"),
+                    bootstrap_address,
                 )
+                cli_logger.newline()
+            if ray_constants.ENABLE_RAY_CLUSTER:
                 cli_logger.print(
-                    "To accept connections from remote hosts, "
-                    "specify a public ip when starting"
+                    "To connect to this Ray cluster, run `ray.init()` as usual:"
                 )
+                with cli_logger.indented():
+                    cli_logger.print("{} ray", cf.magenta("import"))
+                    cli_logger.print(
+                        "ray{}init()",
+                        cf.magenta("."),
+                    )
+                cli_logger.newline()
                 cli_logger.print(
-                    "the head node: ray start --head --node-ip-address=<public-ip>."
+                    "To connect to this Ray instance from outside of "
+                    "the cluster, for example "
                 )
-            cli_logger.newline()
-            cli_logger.print("Alternatively, use the following Python code:")
-            with cli_logger.indented():
-                cli_logger.print("{} ray", cf.magenta("import"))
-                # Note: In the case of joining an existing cluster using
-                # `address="auto"`, the _node_ip_address parameter is
-                # unnecessary.
                 cli_logger.print(
-                    "ray{}init(address{}{}{})",
-                    cf.magenta("."),
-                    cf.magenta("="),
-                    cf.yellow("'auto'"),
-                    ", _node_ip_address{}{}".format(
-                        cf.magenta("="), cf.yellow("'" + node_ip_address + "'")
+                    "when connecting to a remote cluster from your laptop, "
+                    "make sure the"
+                )
+                cli_logger.print(
+                    "dashboard {}is accessible and use Ray jobs. For example:",
+                    f"({dashboard_url}) " if dashboard_url else "",
+                )
+                if dashboard_url:
+                    cli_logger.print(
+                        cf.bold(
+                            "  RAY_ADDRESS='http://<dashboard URL>' ray job submit "
```

Comment:
```suggestion
                            "  RAY_ADDRESS='http://<dashboard URL>:8265' ray job submit "
```
Not sure if `8265` is technically/customarily part of the URL, but I could see a lot of users leaving it out if we don't include it here

Reply:
Or since we're in the `if dashboard_url`, could we use `{dashboard_url}` itself?

Reply:
Ah thanks for the heads up! Yeah actually I tried `dashboard_url` at first, but it defaults to localhost, which is a bit confusing since this part is supposed to be about connecting from a remote client. The `{dashboard_url}` version of the message won't work out of the box in that scenario.

I can also add the actual dashboard port that was used to this message, by the way.